src/sinks/loki/sink.rs

```diff
@@ -92,9 +91,12 @@ impl RequestBuilder<(PartitionKey, Vec<LokiRecord>)> for LokiRequestBuilder {
     type Error = RequestBuildError;
 
     fn compression(&self) -> Compression {
-        match self.compression {
-            CompressionConfigAdapter::Original(compression) => compression,
-            CompressionConfigAdapter::Extended(_) => Compression::None,
+        if self.compression == Compression::Snappy {
+            // Snappy compression is applied after converting the batch to protobuf so
+            // we need to handle this separately.
+            Compression::None
+        } else {
+            self.compression
         }
```

Comment:
Does it make more sense to remove the snap encoding [here](https://github.com/vectordotdev/vector/blob/c9370d61e49686b6f8dfef33d30a37d0ea3e9ef8/lib/loki-logproto/src/lib.rs#L58) and defer to the request builder?